src/client.ts

```diff
@@ -159,4 +165,16 @@ export class BloockClient {
   public async verifySignatures(records: Record[]): Promise<boolean> {
     return this.proofService.verifySignatures(records)
   }
+
+  public async generateSecretKey(): Promise<string> {
```

Comment:
The encryption functions should be included in the Record entity, as you are encrypting a Record